gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Lists;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A class representing a change. A change is an atomic update to a collection of {@link DnsRecord}s
+ * within a {@code ManagedZone}.
+ *
+ * @see <a href="https://cloud.google.com/dns/api/v1/changes">Google Cloud DNS documentation</a>
+ */
+public class ChangeRequest implements Serializable {
+
+  private static final long serialVersionUID = 201601251649L;
+  private final List<DnsRecord> additions;
+  private final List<DnsRecord> deletions;
+  private final String id;
+  private final Long startTimeMillis;
+  private final Status status;
+
+  /**
+   * This enumerates the possible states of a {@code ChangeRequest}.
+   *
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes#resource">Google Cloud DNS
+   * documentation</a>
+   */
+  public enum Status {
+    PENDING("pending"),
+    DONE("done");
+
+    private final String status;
+
+    Status(String status) {
+      this.status = status;
+    }
+
+    static Status translate(String status) {
+      if ("pending".equals(status)) {
```

Comment:
I think you should either assume enum name is always equals to string value lowercased or use a similar pattern
to [that](https://github.com/GoogleCloudPlatform/gcloud-java/blob/e3dba5e9ffcb2bf812af8155f9c846aa1d274940/gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/ValueType.java).